Lib/test/test_zipimport.py

```diff
@@ -239,12 +239,9 @@ def testBadMagic2(self):
         badmagic_pyc = bytearray(test_pyc)
```

Comment:
we should also add an additional `test_bad_pyc_hash` that is like a combination of `test_checked_hash_based_change_pyc` and this `testBadMagic2` where instead of flipping a bit in the magic number, we flip a bit in the pyc hash and expect the error raised by `_bootstrap_external._validate_hash_pyc` to make its way to us.

Reply:
Indeed.

Reply:
Actually the user will never see this exception because the check only happens when the zipfile has a .py file to compare the hash with. And then, if the hash doesn't match it just loads the '.py'.

I can get the exception in a test by temporarily removing the '.py' loader. Is this going too much into internals?
I added this to the end of test_hash_cashed_change_pyc:

        # now remove the '.py' importer so that the '.pyc'
        # failure propagates the exception
        del sys.modules[TESTMOD]
        prev_zip_searchorder = zipimport._zip_searchorder
        zipimport._zip_searchorder = zipimport._zip_searchorder[0:3]
        try:
            self.doTest(None, files, TESTMOD)
            raise AssertionError("This should not be reached")
        except zipimport.ZipImportError as exc:
            cause = "hash in bytecode doesn't match hash of source"
            cause = f"{cause} {TESTMOD!r}"
            expected_msg = f"module load failed: {cause}" 
            self.assertIn(expected_msg, exc.msg)
            self.assertIsInstance(exc.__cause__, ImportError)
            self.assertIn(cause, exc.__cause__.msg)
        finally:
            zipimport._zip_searchorder = prev_zip_searchorder